community/mtd-utils/APKBUILD

```diff
@@ -0,0 +1,122 @@
+# Maintainer: Olliver Schinagl <oliver@schinagl.nl>
+pkgname=mtd-utils
+pkgver=2.1.0
+pkgrel=0
+pkgdesc="Utilities for handling MTD devices, and for dealing with FTL, NFTL JFFS2, etc."
+url="http://www.linux-mtd.infradead.org/"
+arch="all"
+license="GPL-2.0-only"
+makedepends="
+	acl-dev
+	cmocka-dev
+	linux-headers
+	lzo-dev
+	openssl-dev
+	util-linux-dev
+	zlib-dev
+"
+subpackages="
+	${pkgname}-doc
+	${pkgname}-flash::all
+	${pkgname}-jffs::all
+	${pkgname}-misc::all
+	${pkgname}-nand::all
+	${pkgname}-nor::all
+	${pkgname}-ubi::all
+"
+source="
+	ftp://ftp.infradead.org/pub/mtd-utils/${pkgname}-${pkgver}.tar.bz2
+	0001_mtd-2.1.0_fix_tests.patch
+"
+
+build()
+{
+	./configure \
+	            --build="${CBUILD}" \
+	            --enable-tests \
+	            --enable-unit-tests \
+	            --host="${CHOST}" \
+	            --mandir="/usr/share/man" \
+	            --prefix="/usr" \
+	            --sysconfdir="/etc"
+	make
+}
+
+check()
+{
+	make check
+}
+
+flash()
+{
+	pkgdesc="MTD utils (flash commands)"
+
+	mkdir -p "${subpkgdir}/usr/sbin"
+	find "${pkgdir}/usr/sbin/" -type f \( \
+	     -iname '*flash*' \)\
+	     -exec mv "{}" "${subpkgdir}/usr/sbin/" +
+}
+
+jffs()
+{
+	pkgdesc="MTD utils (JFFS* commands)"
+
+	echo "${subpkgdir}"
+	mkdir -p "${subpkgdir}/usr/sbin"
+	find "${pkgdir}/usr/sbin/" -type f \(\
+	     -iname '*jffs*' -o -iname 'sumtool' \)\
+	     -exec mv "{}" "${subpkgdir}/usr/sbin/" +
+}
+
+misc()
+{
+	pkgdesc="MTD utils (miscellaneous commands)"
+
+	mkdir -p "${subpkgdir}/usr/sbin"
+	find "${pkgdir}/usr/sbin/" -type f \(\
+	     -iname '*doc*' -o -iname 'ftl*' -o \
+	     -iname 'mtdpart' -o -iname 'mtd_debug' -o \
+	     -iname 'recv_image' -o -iname 'serve_image' \)\
+	     -exec mv "{}" "${subpkgdir}/usr/sbin/" +
+}
+
+nand()
+{
+	pkgdesc="MTD utils (NAND and FTL commands)"
+
+	mkdir -p "${subpkgdir}/usr/sbin"
+	find "${pkgdir}/usr/sbin/" -type f \(\
+	     -iname 'nand*' -o -iname 'nftl*' \)\
+	     -exec mv "{}" "${subpkgdir}/usr/sbin/" +
+}
+
+nor()
+{
+	pkgdesc="MTD utils (NOR flash commands)"
+
+	mkdir -p "${subpkgdir}/usr/sbin"
+	find "${pkgdir}/usr/sbin/" -type f \(\
+	     -iname '*nor*' -o -iname 'rfd*' \)\
+	     -exec mv "{}" "${subpkgdir}/usr/sbin/" +
+}
+
+ubi()
+{
+	pkgdesc="MTD utils (UBI and UBIFS commands)"
+
+	mkdir -p "${subpkgdir}/usr/sbin"
+	find "${pkgdir}/usr/sbin/" -type f \(\
+	     -iname '*ubi*' -o \
+	     -iname 'lsmtd' -o -iname 'mtdinfo' \)\
+	     -exec mv "{}" "${subpkgdir}/usr/sbin/" +
+}
+
+package()
+{
+	depends="${subpackages##*-doc}"
```

Comment:
Only non-style issue I have is that this relies on the doc package being listed as first in the subpackages list, which might be non-obvious for others and can hide subpackages that are mentioned before.

I think the best way to handle this is to have a separate variable containing all the subpackages the main package depends on, and then include that both in the subpackages and depends variable.

Reply:
fair enough; to be fair, I didnt' even properly test it either, so shame on me here. I just typed it; and wanted to check if it would remove it from any random spot. (and for any -doc package)

So if you prefer to have _subpackages I'll do that instead.